### PR TITLE
Use UseCases For Mock API

### PR DESCRIPTION
**Pull request by Nephelite**
Other changes:
 - Isolated the Category Filter panel into its own component file.
 - Refactored Breadcrumb to not use Breadcrumb.Item because it's deprecated.
 - Refactored Dropdown to not use the overlay prop because that's also deprecated.

### CODE REVIEW

File: frontend/src/presentation/components/Category/CategoryFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    const handleCategoryToDeleteChange = (category: string, checked: boolean) => {
+        const updatedCategories = checked
+            ? [...categoriesToDelete, category]
+            : categoriesToDelete.filter((c) => c !== category);
+        setCategoriesToDelete(updatedCategories);
+    };
+
+    const confirmDeletion = () => {
+        if (categoriesToDelete.length === 0) {
+            Modal.info({
```
Comment (njxue) [inline frontend/src/presentation/components/Category/CategoryFilter.tsx:65]: Maybe consider disabling the delete button instead (is it possible?)

File: frontend/src/presentation/components/QuestionFilters.tsx [hunk 2]
```diff
@@ -36,93 +36,90 @@ export const QuestionFilters: React.FC<QuestionFiltersProps> = ({
...
+            message.success("Category added successfully!");
+        } catch (error) {
+            message.error("Failed to add category!");
+            console.error("Failed to add category:", error);
+        }
+    };
+
+    const handleDeleteCategory = async (categoriesToDelete: string[]) => {
+        try {
+            await Promise.all(categoriesToDelete.map((category) => categoryUseCases.deleteCategory(category)));
```
Comment (njxue) [inline frontend/src/presentation/components/QuestionFilters.tsx:93]: I think we need BE to have bulk deletion too haha. Can deprioritise though

**Review by njxue — APPROVED** [review]

File: frontend/src/presentation/pages/QuestionsPage.tsx [hunk 3]
```diff
@@ -71,11 +71,11 @@ const QuestionsPage: React.FC = () => {
         }
     };
 
-    const handleBreadcrumbClick = (item: string) => () => {
+    const handleBreadcrumbClick = useCallback((item: string) => () => {
```
Comment (njxue) [inline frontend/src/presentation/pages/QuestionsPage.tsx]: Hmm not sure why this useCallback is required

File: frontend/src/presentation/components/QuestionFilters.tsx (hunk 2 above)
Comment (Nephelite) [reply frontend/src/presentation/components/QuestionFilters.tsx:93]: no need lah, a bit too troublesome, and it isn't expected for the deletion to have too many entries

File: frontend/src/presentation/pages/QuestionsPage.tsx (hunk 3 above)
Comment (Nephelite) [reply frontend/src/presentation/pages/QuestionsPage.tsx]: Fair enough

File: frontend/src/presentation/components/Category/CategoryFilter.tsx (hunk 1 above)
Comment (Nephelite) [reply frontend/src/presentation/components/Category/CategoryFilter.tsx:65]: It's already disabled, this is just a just in case practice

**Pull request merged**